libnavigation-core/src/main/java/com/mapbox/navigation/core/fasterroute/FasterRouteController.kt

```diff
@@ -0,0 +1,70 @@
+package com.mapbox.navigation.core.fasterroute
+
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import com.mapbox.api.directions.v5.models.RouteOptions
+import com.mapbox.navigation.base.extensions.ifNonNull
+import com.mapbox.navigation.core.directions.session.DirectionsSession
+import com.mapbox.navigation.core.directions.session.RoutesRequestCallback
+import com.mapbox.navigation.core.trip.session.TripSession
+import com.mapbox.navigation.utils.timer.MapboxTimer
+import java.util.concurrent.CopyOnWriteArrayList
+
+internal class FasterRouteController(
```

Comment:
Should we make this `object`?

Reply:
Personally like to avoid `object` if we can. I was bit by the global testing issues recently https://github.com/mapbox/mapbox-navigation-android/pull/2553

Reply:
Yeah I agree. Singletons and tests don't get along 😅 We do have `mockkObject` in MockK yes, but if you forget, you run into issues as @kmadsen described (also IMO singletons shouldn't be mocked at all, so that solution is a hack 😬).